packages/alice/lib/core/alice_logger.dart

```diff
@@ -3,76 +3,48 @@ import 'dart:io' show Process, ProcessResult;
 import 'package:alice/helper/operating_system.dart';
 import 'package:alice/model/alice_log.dart';
 import 'package:alice/utils/num_comparison.dart';
-import 'package:flutter/foundation.dart';
+import 'package:rxdart/rxdart.dart';
 
 /// Logger used to handle logs from application.
 class AliceLogger {
-  AliceLogger({int? maximumSize = 1000}) : _maximumSize = maximumSize;
+  /// Maximum logs size. If 0, logs will be not rotated.
+  final int maximumSize;
 
-  final ValueNotifier<List<AliceLog>> _logs = ValueNotifier<List<AliceLog>>([]);
+  /// Subject which keeps logs.
+  final BehaviorSubject<List<AliceLog>> _logsSubject;
 
-  ValueListenable<List<AliceLog>> get listenable => _logs;
+  AliceLogger({required this.maximumSize})
+      : _logsSubject = BehaviorSubject.seeded([]);
 
-  List<AliceLog> get logs => listenable.value;
+  /// Getter of stream of logs
+  Stream<List<AliceLog>> get logsStream => _logsSubject.stream;
 
-  int? _maximumSize;
+  /// Getter of all logs
+  List<AliceLog> get logs => _logsSubject.value;
 
-  /// The maximum number of logs to store or `null` for unlimited storage.
-  ///
-  /// If more logs arrive, the oldest ones (based on their [
-  /// AliceLog.timestamp]) will be removed.
-  int? get maximumSize => _maximumSize;
-
-  set maximumSize(int? value) {
-    _maximumSize = maximumSize;
-
-    if (value != null && logs.length > value) {
-      _logs.value = logs.sublist(logs.length - value, logs.length);
-    }
-  }
-
-  void addAll(List<AliceLog> logs) {
+  /// Adds all logs.
+  void addAll(Iterable<AliceLog> logs) {
     for (var log in logs) {
       add(log);
     }
   }
 
+  /// Add one log. It sorts logs after adding new element. If [maximumSize] is
+  /// set and max size is reached, first log will be deleted.
   void add(AliceLog log) {
-    late final int index;
-    if (logs.isEmpty || !log.timestamp.isBefore(logs.last.timestamp)) {
-      // Quick path as new logs are usually more recent.
-      index = logs.length;
-    } else {
-      // Binary search to find the insertion index.
-      int min = 0;
-      int max = logs.length;
-      while (min < max) {
-        final int mid = min + ((max - min) >> 1);
-        final AliceLog item = logs[mid];
-        if (log.timestamp.isBefore(item.timestamp)) {
-          max = mid;
-        } else {
-          min = mid + 1;
-        }
-      }
-      assert(min == max, '');
-      index = min;
+    final values = _logsSubject.value;
+    final count = values.length;
+    if (maximumSize > 0 && count.gte(maximumSize)) {
```

Comment:
Using `.gte` is not needed any longer since `maximumSize` is no longer nullable, so you should change it back to `>=`.